boot/assets.go

```diff
@@ -274,8 +274,8 @@ type TrustedAssetsInstallObserver struct {
 // measured as part of the secure boot or the bootloader configuration.
 //
 // Implements gadget.ContentObserver.
-func (o *TrustedAssetsInstallObserver) Observe(op gadget.ContentOperation, affectedStruct *gadget.LaidOutStructure, root, relativeTarget string, data *gadget.ContentChange) (gadget.ContentChangeAction, error) {
-	if affectedStruct.Role() != gadget.SystemBoot {
+func (o *TrustedAssetsInstallObserver) Observe(op gadget.ContentOperation, partRole string, root, relativeTarget string, data *gadget.ContentChange) (gadget.ContentChangeAction, error) {
```

Comment:
nitpick; you could leave out the string type and chain with the root ..., same in other similar places

Reply:
Good catch, done